src/widget/wcollapsiblegroupbox.cpp

```diff
@@ -0,0 +1,107 @@
+#include "widget/wcollapsiblegroupbox.h"
+
+#include <QEvent>
+#include <QStyleOption>
+#include <QStylePainter>
+
+#include "moc_wcollapsiblegroupbox.cpp"
+#include "util/assert.h"
+
+void CollapsibleGroupBoxStyle::drawPrimitive(QStyle::PrimitiveElement element,
+        const QStyleOption* pOption,
+        QPainter* pPainter,
+        const QWidget* pWidget) const {
+    const WCollapsibleGroupBox* pCGB = qobject_cast<const WCollapsibleGroupBox*>(pWidget);
+    if (pCGB && pCGB->isCheckable()) {
+        if (element == QStyle::PE_IndicatorCheckBox) {
+            // Draw Down/Right arrow instead of original checkbox
+            QProxyStyle::drawPrimitive(pCGB->isChecked()
+                            ? QStyle::PE_IndicatorArrowDown
+                            : QStyle::PE_IndicatorArrowRight,
+                    pOption,
+                    pPainter,
+                    pWidget);
+            return;
+        } else if (!pCGB->isChecked() && element == QStyle::PE_FrameGroupBox) {
+            // When the box is collapsed, the default style might still draw the
+            // frame, e.g. underneath the title, even though the box is reduced
+            // to its title row, so skip drawing the frame.
+            return;
+        }
+    }
+
+    QProxyStyle::drawPrimitive(element, pOption, pPainter, pWidget);
+}
+
+WCollapsibleGroupBox::WCollapsibleGroupBox(QWidget* pParent)
+        : QGroupBox(pParent),
+          m_minH(-1),
+          m_maxH(10000) {
+    connect(this,
+            &QGroupBox::toggled,
+            this,
+            &WCollapsibleGroupBox::slotToggled);
+    // Set the custom style for the expand/collapse icons and other tweaks
+    setStyle(new CollapsibleGroupBoxStyle(style()));
+    setFocusPolicy(Qt::NoFocus);
+}
+
+WCollapsibleGroupBox::WCollapsibleGroupBox(const QString& title, QWidget* pParent)
+        : WCollapsibleGroupBox(pParent) {
+    setTitle(title);
+}
+
+bool WCollapsibleGroupBox::event(QEvent* pEvent) {
+    if (isCheckable()) {
```

Comment:
Could you please refactor some of these nested condition to flatten it and use early return instead? It is a bit hard to follow at the moment

Reply:
Sure, I'll try.

Reply:
```suggestion
    if (!isCheckable()) {
        QGroupBox::event(pEvent);
        return;
    }
```

Reply:
Yes, that's the only one.